api/C/include/csi_ll_pmp.h

```diff
@@ -0,0 +1,237 @@
+/*
+ * Low-Level PMP API
+ *
+ * This module provides functionality for PMP configuration.
+ *
+ * Copyright (C) 2017-2023 Alibaba Group Holding Limited
+ */
+
+#ifndef CSI_LL_PMP_H
+#define CSI_LL_PMP_H
+
+#include "csi_types.h"
+#include <stdint.h>
+
+/*
+ * Types of address Matching
+ */
+typedef enum {
+    ADDRESS_MATCHING_OFF   = 0,	/* Null region (disabled) */
+    ADDRESS_MATCHING_TOR   = 1, /* Top of range */
+    ADDRESS_MATCHING_NA4   = 2, /* Naturally aligned four-byte region */
+    ADDRESS_MATCHING_NAPOT = 3, /* Naturally aligned power-of-two region, ≥8 bytes */
+} csi_addr_matching_t;
+
+/*
+ * Attribution of PMP coniguration
+ */
+typedef struct {
+    bool flag_r; /* The flag of readable. 0: not readable; 1: readable */
```

Comment:
Here I prefer to list these flags_r/w/x/lock into one variable which match the PMP configuration register format, so user can easily set the pmpcfg using this flags value without complex if else calculation.

1. ![image](https://github.com/riscv-non-isa/riscv-rvm-csi/assets/1538922/c1bbb4b8-435e-461c-a97f-ff55feb4593b)
2. The generated encoding.h also contains pmp attribute macros which can be easily used https://github.com/riscv/riscv-opcodes/blob/master/encoding.h#L276-L285

Reply:
@baiqiang728 Using bit flag to operate with such PMP flag is best choice.

Reply:
I agree - then have one function to set the configuration instead of lots of little functions dealing with each flag.

Reply:
For the moment though, define our own enumeration rather than trying to include encodings.h.  We will auto-generate a similar file to encodings.h but this will happen later.

Reply:
fixed.
I've changed to one function to set the configuration instead of lots of little functions dealing with each flag